lib/process/teleport.go

```diff
@@ -139,3 +144,19 @@ func (p *Process) getAuthGatewayConfig() (storage.AuthGateway, error) {
 	}
 	return opsservice.GetAuthGateway(client, p.identity)
 }
+
+// proxySettings returns Teleport proxy settings based on the Teleport config.
+func (p *Process) proxySettings() client.ProxySettings {
+	settings := client.ProxySettings{
+		Kube: client.KubeProxySettings{
+			Enabled: p.teleportConfig.Proxy.Kube.Enabled,
+		},
+		SSH: client.SSHProxySettings{
+			ListenAddr: p.teleportConfig.Proxy.SSHAddr.String(),
+		},
+	}
+	if len(p.teleportConfig.Proxy.Kube.PublicAddrs) > 0 {
+		settings.Kube.PublicAddr = p.teleportConfig.Proxy.Kube.PublicAddrs[0].String()
```

Comment:
Have you considered shortening this to:
```go
settings.Kube.PublicAddr = p.teleportConfig.FirstProxyPublicKubeAddr()
```
?

Reply:
This is teleport struct, it would be nice if it had this method but not sure it's worth the trouble adding it specifically for this PR.